### PR TITLE
Add more PriorityClassName fields in Helm charts

### LINKED ISSUE #10504: Set priority class based on k8s version
Priority class does not require to be on kube-system since 1.xx.

1. Check which kubernetes versions this requirement was lifted
2. Apply logic to add the priority class depending on the k8s version and the namespace is being installed.

More info: https://github.com/kubernetes/kubernetes/pull/76310
More info: https://github.com/kubernetes/kubernetes/issues/76308

### PR DESCRIPTION
**Pull request by johngmyers**
Set `priorityClassName` when installing outside the `kube-system` namespace on Kubernetes 1.17 or higher. Add some `priorityClassName` fields that were missing.

Fixes #10504 

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/10583)

### CODE REVIEW

**Review by tgraf — APPROVED** [review]

**Comment by tgraf** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Comment by joestringer** [comment]
VM provisioning issue:
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/17965/execution/node/114/log/

**Comment by joestringer** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by christarazi** [comment]
Build expired. Triggering...

**Comment by christarazi** [comment]
test-me-please

File: install/kubernetes/cilium/charts/agent/templates/daemonset.yaml [hunk 1]
```diff
@@ -307,7 +307,7 @@ spec:
           {{- toYaml .Values.initResources | trim | nindent 10 }}
 {{- end }}
       restartPolicy: Always
-{{- if and (eq .Release.Namespace "kube-system") (or (gt .Capabilities.KubeVersion.Minor "10") (gt .Capabilities.KubeVersion.Major "1"))}}
+{{- if or (and (eq .Release.Namespace "kube-system") (gt .Capabilities.KubeVersion.Minor "10")) (gt .Capabilities.KubeVersion.Minor "16") (gt .Capabilities.KubeVersion.Major "1"))}}
```
Comment (aanm) [inline install/kubernetes/cilium/charts/agent/templates/daemonset.yaml]: There's a missing `and` in the end, it should be like this:
`{{- if or (and (eq .Release.Namespace "kube-system") (gt .Capabilities.KubeVersion.Minor "10"))  (and (gt .Capabilities.KubeVersion.Minor "16") (gt .Capabilities.KubeVersion.Major "1"))}}`

**Review by aanm — CHANGES_REQUESTED** [review]

File: install/kubernetes/cilium/charts/agent/templates/daemonset.yaml (hunk 1 above)
Comment (tgraf) [reply install/kubernetes/cilium/charts/agent/templates/daemonset.yaml]: This is wrong as well. It must be `or (gt .Capabilities.KubeVersion.Minor "16") (gt .Capabilities.KubeVersion.Major "1")`. The idea is to match on 2.x, 3.x, 4.x, ... as well.

Comment (aanm) [reply install/kubernetes/cilium/charts/agent/templates/daemonset.yaml]: @johngmyers
`{{- if or (and (eq .Release.Namespace "kube-system") (gt .Capabilities.KubeVersion.Minor "10")) (or (gt .Capabilities.KubeVersion.Minor "16") (gt .Capabilities.KubeVersion.Major "1"))}}`

Comment (johngmyers) [reply install/kubernetes/cilium/charts/agent/templates/daemonset.yaml]: "or" is variadic. The problem was an extra close parenthesis.

**Comment by aanm** [comment]
test-me-please

File: install/kubernetes/cilium/charts/agent/templates/daemonset.yaml (hunk 1 above)
Comment (aanm) [reply install/kubernetes/cilium/charts/agent/templates/daemonset.yaml]: @johngmyers the way it is right now won't work for k8s 2.x, 3.x like Thomas point it out.

Comment (johngmyers) [reply install/kubernetes/cilium/charts/agent/templates/daemonset.yaml]: Why wouldn't it work for k8s 2.x or 3.x? The `(gt .Capabilities.KubeVersion.Major "1")` clause takes care of those cases.

I had only fixed one of the two extraneous close parens. Fixed the other one.

Comment (aanm) [reply install/kubernetes/cilium/charts/agent/templates/daemonset.yaml]: sorry I misread the last change. Now seems to be good.

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Comment by christarazi** [comment]
test-me-please

File: install/kubernetes/cilium/charts/agent/templates/daemonset.yaml [hunk 2]
```diff
@@ -307,7 +307,7 @@ spec:
           {{- toYaml .Values.initResources | trim | nindent 10 }}
 {{- end }}
       restartPolicy: Always
-{{- if and (eq .Release.Namespace "kube-system") (or (gt .Capabilities.KubeVersion.Minor "10") (gt .Capabilities.KubeVersion.Major "1"))}}
+{{- if or (and (eq .Release.Namespace "kube-system") (gt .Capabilities.KubeVersion.Minor "10")) (gt .Capabilities.KubeVersion.Minor "16") (gt .Capabilities.KubeVersion.Major "1")}}
```
Comment (aanm) [inline install/kubernetes/cilium/charts/agent/templates/daemonset.yaml:310]: @johngmyers this is not in the correct format. (same for all other changes)

It should be:
`{{- if or (and (eq .Release.Namespace "kube-system") (gt .Capabilities.KubeVersion.Minor "10")) (or (gt .Capabilities.KubeVersion.Minor "16") (gt .Capabilities.KubeVersion.Major "1"))}}`

**Review by aanm — CHANGES_REQUESTED** [review]

File: install/kubernetes/cilium/charts/agent/templates/daemonset.yaml (hunk 2 above)
Comment (johngmyers) [reply install/kubernetes/cilium/charts/agent/templates/daemonset.yaml:310]: Fixed two more extraneous close parentheses.

The or operator in Go template is variadic, so the second or operator in your suggestion is unnecessary.

**Comment by aanm** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Comment by johngmyers** [comment]
Test failure looks like a flake?

**Comment by aanm** [comment]
yes it is @johngmyers . Thanks for your patience for all the requested changes!

**Pull request merged**